src/Akka.Hosting/Configuration/HoconConfigurationSection.cs

```diff
@@ -0,0 +1,83 @@
+﻿// -----------------------------------------------------------------------
+//   <copyright file="HoconConfigurationSection.cs" company="Petabridge, LLC">
+//     Copyright (C) 2015-2023 .NET Petabridge, LLC
+//   </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Primitives;
+
+namespace Akka.Hosting.Configuration;
+
+internal class HoconConfigurationSection: IConfigurationSection
+{
+    public static HoconConfigurationSection Create(IConfigurationSection section)
+    {
+        if (section is HoconConfigurationSection hoconSection)
+            return hoconSection;
+
+        if (section is not ConfigurationSection configSection)
+            throw new Exception($"{nameof(section)} argument could not be cast to {nameof(ConfigurationSection)}");
+
+        var fieldInfo = typeof(ConfigurationSection).GetField("_root", BindingFlags.Instance | BindingFlags.NonPublic);
```

Comment:
We need to do some reflection magic here to retrieve the original root configuration providers. This needs to be done because these roots are the actual origin of all configuration values, any environment variable provider need to be replaced with our own provider.